tools/astarte_dev_tool/lib/utilities/auth.ex

```diff
@@ -18,8 +18,29 @@
 
 defmodule AstarteDevTool.Utilities.Auth do
   alias Astarte.Client.Credentials
+  alias X509.PrivateKey
+  alias X509.PublicKey
 
   def gen_auth_token(private_key) when is_bitstring(private_key) do
     Credentials.dashboard_credentials() |> Credentials.to_jwt(private_key)
   end
+
+  def pem_key(), do: {:ok, PrivateKey.new_ec(:secp256r1) |> PrivateKey.to_pem()}
+
+  def pem_key(priv) do
+    case PrivateKey.from_pem(priv) do
+      {:ok, result} -> {:ok, result |> PublicKey.derive() |> PublicKey.to_pem()}
+      error -> error
+    end
+  end
```

Comment:
I'm not a fan of using the same approach as the CLI here. It is not clear what the `pem_key` function does. One option could be to use an option (pun intended) to pass in the private key, something like
```elixir
def pem_key(opts \\ []) do
case Keyword.fetch(:private_key) do
#...
```
but in my opinion it would be clearer to have two different functions, one for generating the private key and another for the public key.

Reply:
I prefer the ‘erlang’ approach. lots of simple functions

Reply:
imho, pem_key could make sense as a single function that does both, but a clearer invocation (so, an explicit field :private_key) surely help to better understand without necessarily checking the code

Like what 
```
OptionParser.parse!(args, strict: @switches, aliases: @aliases)
```
does with `strict:` and `aliases:`

Reply:
There is no way for a reader to understand what `pem_key` does without reading the code.

The `X509` library handles [private](https://hexdocs.pm/x509/X509.PrivateKey.html) and [public](https://hexdocs.pm/x509/X509.PublicKey.html) keys in two different modules (modules! not even functions); the same approach is used by Erlang's [`public_key`](https://www.erlang.org/doc/apps/public_key/public_key.html) library, while the [`:crypto`](https://www.erlang.org/doc/apps/crypto/crypto#key-api) library outputs both the private and public key in just one functions, and documents the fact that you can pass a private key as an optional argument using typespecs.

The current approach is suitable for a CLI command, but too opaque for implementation code. In my opinion, the way to go is using two functions, e.g. `new_ec_private_key/0` and `public_key_from/1`, but a number of different options is available.

Reply:
Although in my opinion the single function was the most elegant solution, it is fine. I used the `atom' approach to solve

Reply:
I am sorry, but this is still not a clear solution. In the same way as before, there is no way for a reader to understand what the function does without looking at implementation details (the atom, in this case).
To move out from this impasse, I suggest either changing function names (since the two functions do two different things) or generating both keys in the same function (the `:crypto` approach)